src/ripple/app/tx/impl/DeleteAccount.cpp

```diff
@@ -0,0 +1,337 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2018 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/tx/impl/DeleteAccount.h>
+#include <ripple/app/tx/impl/DepositPreauth.h>
+#include <ripple/app/tx/impl/SetSignerList.h>
+#include <ripple/basics/Log.h>
+#include <ripple/basics/mulDiv.h>
+#include <ripple/protocol/Feature.h>
+#include <ripple/protocol/Indexes.h>
+#include <ripple/protocol/st.h>
+#include <ripple/protocol/TxFlags.h>
+#include <ripple/ledger/View.h>
+
+namespace ripple {
+
+NotTEC
+DeleteAccount::preflight (PreflightContext const& ctx)
+{
+    if (! ctx.rules.enabled(featureDeletableAccounts))
+        return temDISABLED;
+
+    if (ctx.tx.getFlags() & tfUniversalMask)
+        return temINVALID_FLAG;
+
+    auto const ret = preflight1 (ctx);
+
+    if (!isTesSuccess (ret))
+        return ret;
+
+    if (ctx.tx[sfAccount] == ctx.tx[sfDestination])
+        // An account cannot be deleted and give itself the resulting XRP.
+        return temDST_IS_SRC;
+
+    return preflight2 (ctx);
+}
+
+std::uint64_t
+DeleteAccount::calculateBaseFee (
+    ReadView const& view,
+    STTx const& tx)
+{
+    // The fee required for AccountDelete is one owner reserve.  But the
+    // owner reserve is stored in drops.  We need to convert it to fee units.
+    Fees const& fees {view.fees()};
+    std::pair<bool, std::uint64_t> const mulDivResult {
+        mulDiv (fees.increment, fees.units, fees.base)};
+    if (mulDivResult.first)
+        return mulDivResult.second;
+
+    // If mulDiv returns false then overflow happened.  Punt by using the
+    // standard calculation.
+    return Transactor::calculateBaseFee (view, tx);
+}
+
+namespace 
+{
```

Comment:
Extra whitespace

Reply:
Thanks for pointing out all the extra whitespace.  This surprised me, since up until recently Sublime Text has stripped trailing whitespace on exit for me.  I believe I've put a fix in my config file now, so I think things will hopefully be better.